docs/csharp/whats-new/csharp-6.md

```diff
@@ -98,483 +36,157 @@ public class Student
-
 ## Await in Catch and Finally blocks
 
-C# 5 had several limitations around where you could place `await` expressions.
-One of those has been removed in C# 6. You can now use `await` in `catch`
-or `finally` expressions. 
-
-The addition of await expressions in catch and finally blocks may appear
-to complicate how those are processed. Let's add an example to discuss
-how this appears. In any async method, you can use an await expression
-in a finally clause.
-
-With C# 6, you can also await in catch expressions. This is most often
-used with logging scenarios:
+C# 5 had several limitations around where you could place `await` expressions. You can now use `await` in `catch` or `finally` expressions.  With C# 6, you can also await in catch expressions. This is most often used with logging scenarios:
 
 [!code-csharp[AwaitFinally](../../../samples/snippets/csharp/new-in-6/NetworkClient.cs#AwaitFinally)]
 
-The implementation details for adding `await` support inside `catch`
-and `finally` clauses ensures that the behavior is consistent with the
-behavior for synchronous code. When code executed in a `catch` or `finally`
-clause throws, execution looks for a suitable `catch` clause in the next
-surrounding block. If there was a current exception, that exception is
-lost. The same happens with awaited expressions in `catch` and `finally`
-clauses: a suitable `catch` is searched for, and the current exception,
-if any, is lost.  
+The implementation details for adding `await` support inside `catch` and `finally` clauses ensure that the behavior is consistent with the behavior for synchronous code. When code executed in a `catch` or `finally` clause throws, execution looks for a suitable `catch` clause in the next surrounding block. If there was a current exception, that exception is lost. The same happens with awaited expressions in `catch` and `finally` clauses: a suitable `catch` is searched for, and the current exception, if any, is lost.  
 
 > [!NOTE]
-> This behavior is the reason it's recommended to write `catch` and `finally`
-> clauses carefully, to avoid introducing new exceptions.
+> This behavior is the reason it's recommended to write `catch` and `finally` clauses carefully, to avoid introducing new exceptions.
 
 ## Index initializers
 
-*Index Initializers* is one of two features that make collection
-initializers more consistent with index usage. In earlier releases of C#, you could use
-*collection initializers* only with sequence style collections, including <xref:System.Collections.Generic.Dictionary%602> by adding braces around key and value pairs:
+*Index Initializers* is one of two features that make collection initializers more consistent with index usage. In earlier releases of C#, you could use *collection initializers* only with sequence style collections, including <xref:System.Collections.Generic.Dictionary%602> by adding braces around key and value pairs:
```

Comment:
@BillWagner I'm afraid such a presentation of index initializers might be understood in a limited way: only in the context of collections. Also mentioning of *collection initializers* doesn't help, because ultimately the feature is about *object initializers*. 

Let's take the [C# 5 specification](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-334.pdf) and check the "12.7.11.3 Object initializers" section (page 154). It starts with the following sentence:
>An _**object initializer**_ specifies values for zero or more fields or properties of an object.

Then, let's check the [corresponding section in the C# 6 specification draft](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#object-initializers). It starts with:
>An _**object initializer**_ specifies values for zero or more fields, properties or indexed elements of an object.

So, it's an update of the *object initializer* feature, and not of the *collection initializers* as one might interpret the following sentence:
>*Index Initializers* is one of two features that make collection initializers more consistent with index usage.

What makes it look like it's about collections is the fact that indexers usually have "collection semantics" with the `Dictionary` class being the most popular example of a class with an indexer. So, an upgrade of an object initializer feature mostly helps to initialize collections with indexers.

Here, I agree with [the comment](https://github.com/dotnet/docs/issues/5169#issuecomment-392382085) of @drewjcooper in the original [issue](https://github.com/dotnet/docs/issues/5169).

Reply:
I agree as well.  However, I think that goes deeper than I think we should on the "what's new" article.

I'm updating this PR to include that discussion in the document on [object initializers](https://docs.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers).  Then, link to it from here.

Reply:
@BillWagner oh...! That linked article about object initializers is indeed outdated. It misses indexers (starting with C# 6):
>Object initializers let you assign values to any accessible fields or properties of an object at creation time ...

Though, again, indexers are in the section about collections:
>You can specify indexed elements if the collection supports indexing.